eslint.config.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import pluginJs from '@eslint/js';
+import noOnlyTests from 'eslint-plugin-no-only-tests';
+import tseslint from 'typescript-eslint';
+
+export default tseslint.config(
+  {
+    ignores: [
+      '**/*.js',
+      '**/*.cjs',
+      '**/*.mjs',
+      '**/*.d.ts',
+      '**/.wireit/',
+      '**/node_modules/',
+      'lib/',
+      'vscode-extension/.vscode-test/',
+      'vscode-extension/lib/',
+      'vscode-extension/built/',
+    ],
+  },
+  {
+    files: ['**/*.ts'],
+    languageOptions: {
+      parser: tseslint.parser,
```

Comment:
I didn't think you had to set the parser manually anymore.

Here's a recent config I made when upgrading to ESLint v9:

```ts
import tseslint from 'typescript-eslint';
import eslint from '@eslint/js';
import eslintPluginPrettierRecommended from 'eslint-plugin-prettier/recommended';

export default [
  eslint.configs.recommended,
  eslintPluginPrettierRecommended,
  ...tseslint.configs.recommendedTypeChecked,
  {
    languageOptions: {
      parserOptions: {
        projectService: true,
        tsconfigRootDir: import.meta.dirname,
      },
    },
    rules: {
      "@typescript-eslint/no-non-null-assertion": "error"
    },
  },
];
```

It doesn't use `tseslint.config()` because the config is in .js and doesn't need a type-safe config merger.

Reply:
> I didn't think you had to set the parser manually anymore.

Huh, you're right! I think I was getting mislead by a combination of things:

- My config was initially matching .js files not covered by a TypeScript config because I didn't have the right `files` filter.
 
- eslint then tried to run a TypeScript-parser-depending rule on that js file, but that rule has an outdated error message saying that you need to set a parser. But the real problem was just that it shouldn't have been covered by the TypeScript rule at all.


> It doesn't use tseslint.config() because the config is in .js and doesn't need a type-safe config merger.

Ah nice. Didn't realize this was just a typings thing.

> Here's a recent config I made when upgrading to ESLint v9:

Cool! I adopted some of this.

The main difference now is that I have this `{files: ["**/*.ts"]}` constraint on all the TypeScript setting objects (including the tseslint ones). What I found is that I otherwise get errors with `projectService: true` where I get errors about tseslint not being able to find a `tsconfig.json` for the one or two `.js` files we have here. A little odd, but I can't find a simpler arrangement after fiddling for a while -- I always end up with some kind of error relating to non-ts  files.

Reply:
>   eslintPluginPrettierRecommended,

Oh also, curious why you like using this prettier eslint plugin? (instead of just using the prettier binary).